x/incentives/keeper/distribute_test.go

```diff
@@ -868,3 +871,196 @@ func (s *KeeperTestSuite) TestGetPoolFromGaugeId() {
+// 7. let epoch 2 pass
+// 		- check only external non-perpetual gauges with 2 epochs distributed
+// 		- check gauge has been correctly updated
+// 8. let epoch 3 pass
+// 		- nothing distributes as non-perpetual gauges with 2 epochs have ended and perpetual gauges have not been reloaded
+//		- nothing should change in terms of incentive records
+func (s *KeeperTestSuite) TestFunctionalInternalExternalCLGauge() {
+	// 1. Initialize variables
+	s.SetupTest()
+
+	requiredBalances := sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(1_000_000_000)), sdk.NewCoin("usdc", sdk.NewInt(1_000_000_000)), sdk.NewCoin("uosmo", sdk.NewInt(1_000_000_000)))
+	internalGaugeCoins := sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(750_000)))                                               // distributed full sum at epoch
+	externalGaugeCoins := sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(1_000_000)), sdk.NewCoin("usdc", sdk.NewInt(1_000_000)))   // distributed full sum at epoch
+	halfOfExternalGaugeCoins := sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(500_000)), sdk.NewCoin("usdc", sdk.NewInt(500_000))) // distributed at each epoch for non-perp gauge with numEpoch = 2
+
+	internalGaugeDecCoins := osmoutils.ConvertCoinsToDecCoins(internalGaugeCoins)
+	externalGaugeDecCoins := osmoutils.ConvertCoinsToDecCoins(externalGaugeCoins)
+	halfOfExternalGaugeDecCoins := osmoutils.ConvertCoinsToDecCoins(halfOfExternalGaugeCoins)
+
+	s.FundAcc(s.TestAccs[0], requiredBalances)
+	s.FundAcc(s.TestAccs[1], requiredBalances)
+	s.FundAcc(s.TestAccs[2], requiredBalances)
+	s.FundModuleAcc(incentivetypes.ModuleName, requiredBalances)
+
+	// 2. Setup CL pool and gauge (gauge automatically gets created at the end of CL pool creation).
+	clPoolId1 := s.PrepareConcentratedPool() // creates internal no-lock gauge id = 1
+	clPoolId2 := s.PrepareConcentratedPool() // creates internal no-lock gauge id = 2
+
+	epochInfo := s.App.IncentivesKeeper.GetEpochInfo(s.Ctx)
+
+	// 3. Create external no-lock gauges for CL pools
+	_, clPoolExternalGaugeIdPool1, clPoolExternalGaugeIdPool2 := s.CreateNoLockExternalGauges(clPoolId1.GetId(), clPoolId2.GetId(), externalGaugeCoins)
+
+	// 4. Create Distribution records to incentivize internal CL no-lock gauges
+	s.IncentivizeInternalGauge(clPoolId1.GetId(), clPoolId2.GetId(), epochInfo.Duration, false)
+
+	// 5. let epoch 1 pass
+	// ******************** EPOCH 1 *********************
+	// let 1 epoch pass
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(epochInfo.Duration))
+	s.App.EpochsKeeper.AfterEpochEnd(s.Ctx, epochInfo.GetIdentifier(), 1)
+
+	// Validate the results
+	clPoolInternalGaugeId, err := s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, clPoolId1.GetId(), epochInfo.Duration)
+	s.Require().NoError(err)
+
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool1, 1, externalGaugeCoins)
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool2, 1, halfOfExternalGaugeCoins)
```

Comment:
Please add a comment why one is half and the other one is full.

Whil I think that comment is sufficient, another way to make this clearer is to also make `CreateNoLockExternalGauge` take `numEpochsPaidOver` as a param. Define `numEpochsPaidOverGaugeTwo` at the top and divide `externalGaugeCoins` by `numEpochsPaidOverGaugeTwo` here

Reply:
added comment as well as changed the variable